### ISSUE TITLE
Can't debug WPF project after upgrade from .NET5 to .NET6

### ISSUE THREAD

**Issue by kvalitetskontrollMarkaryd** [dotnet/wpf#5677]
* .NET Core Version: 6.0.100
* Windows version: Windows 11 Pro
* IDE: Visual Studio 2022 and Rider
* Does the bug reproduce also in WPF for .NET Framework 4.8?: Have not tried but it's working in .NET5.

 **Problem description:**
 Changed from .NET5 to .NET6 by changing TargetFramework in .csproj. I now get the error message in VS2022:
"Unable to attach to CoreCLR. Operation not supported. Unknown error: 0x80004005."
Publish and "Start without debugging" works, both in VS and Rider.

What I have tried:
1. I have uppdated all nugets
2. Re-installed Visual Studio 2022
3. Re-installed .Net6
4. Clean and rebuild.

 **Actual behavior:** 
Visual Studio error message:
![Visual Studio error message](https://user-images.githubusercontent.com/40342638/141107226-3215f1bb-9b80-4459-a386-204b9a8784d5.png)

Rider error message:
C:\Users\user\AppData\Local\JetBrains\Toolbox\apps\Rider\ch-0\212.5284.64\lib\ReSharperHost\JetBrains.Debugger.Worker64c.exe --mode=client --frontend-port=53344 --plugins=C:\Users\user\AppData\Local\JetBrains\Toolbox\apps\Rider\ch-0\212.5284.64\plugins\rider-unity\dotnetDebuggerWorker
One or more errors have occurred.

 **csproj**
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net6.0-windows</TargetFramework>
    <UseWPF>true</UseWPF>
    <RuntimeIdentifier>win10-x64</RuntimeIdentifier>
    <PublishSingleFile>true</PublishSingleFile>
    <ApplicationIcon>Business_Target.ico</ApplicationIcon>
    <UseWindowsForms>False</UseWindowsForms>
    <ImplicitUsings>disable</ImplicitUsings>
	<SelfContained>true</SelfContained>
	<IncludeAllContentForSelfExtract>true</IncludeAllContentForSelfExtract>
  </PropertyGroup>

  <ItemGroup>
    <ProjectReference Include="..\ProjectOrganizer.Dto\ProjectOrganizer.Dto.csproj" />
    <ProjectReference Include="..\ProjectOrganizer.Shared\ProjectOrganizer.Shared.csproj">
    </ProjectReference>
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="AsyncAwaitBestPractices" Version="6.0.2" />
    <PackageReference Include="Autofac" Version="6.3.0" />
    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="7.2.0" />
    <PackageReference Include="DevExpress.Images" Version="21.2.3" />
    <PackageReference Include="DevExpress.Wpf" Version="21.2.3" />
    <PackageReference Include="DevExpress.Wpf.Themes.Office2019Black" Version="21.2.3" />
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.Http.Polly" Version="6.0.0" />
    <PackageReference Include="Mono.Cecil" Version="0.11.4" />
    <PackageReference Include="Polly.Extensions.Http" Version="3.0.0" />
    <PackageReference Include="Prism.Core" Version="8.1.97" />
  </ItemGroup>

</Project>
```

**Comment by lindexi** [comment]
@kvalitetskontrollMarkaryd How about clean the `bin` and `obj` folder?

**Comment by kvalitetskontrollMarkaryd** [comment]
@lindexi Thanks for your reply!
I have tried that with no success. Tried again just now and have the same error still

1. Deleted bin & obj for all projects in the solution.
2. dotner restore.
3. Rebuild solution.

Yesterday I updated an Avalonia testing project from .NET5 to .NET6 the same way, just changed the TargetFramework. I got the same error message that I got on this project, but with Rider. A dotnet restore and rebuild fixed it on that testing project.
"Unable to attach to CoreCLR. Operation not supported. Unknown error: 0x80004005."

**Comment by lindexi** [comment]
@kvalitetskontrollMarkaryd  Where is your dotnet install path?

**Comment by kvalitetskontrollMarkaryd** [comment]
@lindexi "C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.0" I guess!?!

**Comment by kvalitetskontrollMarkaryd** [comment]
System.Private.CoreLib.dll is in that folder.

**Comment by Reelix** [comment]
If your project is set to PublishSingleFile, this issue is being tracked over at https://github.com/dotnet/runtime/issues/45382

**Comment by kvalitetskontrollMarkaryd** [comment]
@Reelix Wow, that helped! Thank's a lot! I did not think that PublishSingleFile would apply to debug at all, only when publish. Thank's!
@lindexi Thank you for your help too!

**Comment by kvalitetskontrollMarkaryd** [comment]
This issue can be marked as solved! Or duplicate [dotnet/runtime#45382](url)
